src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -424,8 +424,17 @@ private static MediaTypeCollection GetDeclaredContentTypes(IApiRequestMetadataPr
                 foreach (var metadataAttribute in responseMetadataAttributes)
                 {
                     metadataAttribute.SetContentTypes(contentTypes);
-
-                    if (metadataAttribute.Type != null)
+                    if (metadataAttribute.Type == typeof(void) && 
+                        type != null &&
+                        (metadataAttribute.StatusCode == StatusCodes.Status200OK || metadataAttribute.StatusCode == StatusCodes.Status201Created))
+                    {
+                        // ProducesResponseTypeAttribute's constructor defaults to setting "Type" to void when no value is specified.
+                        // In this event, use the action's return type for 200 or 201 status codes. This lets you decoarate an action with a
```

Comment:
decoarate  😆

Reply:
This should also include 202

Reply:
So I pointed out the issue with 202 in the issue (https://github.com/aspnet/Mvc/issues/7871#issuecomment-398144056). But essentially there are patterns where 202 is used as a place holder for actions which later respond with a 200 once the result is ready.

The specs also suggest that it's not meant to return the actual thing. From https://tools.ietf.org/html/rfc7231#section-6.3.3

>    The 202 response is intentionally noncommittal.  Its purpose is to
   allow a server to accept a request for some other process (perhaps a
   batch-oriented process that is only run once per day) without
   requiring that the user agent's connection to the server persist
   until the process is completed.  The representation sent with this
   response ought to describe the request's current status and point to
   (or embed) a status monitor that can provide the user with an
   estimate of when the request will be fulfilled.

Seems appropriate to exclude it here.

Reply:
Is that always the same action?

Reply:
Seems like a pattern used commonly with Azure APIs too: https://github.com/jeffhollan/LogicAppsAsyncResponseSample \ https://docs.microsoft.com/en-us/rest/api/hdinsight/asynchronous-operations--202-accepted-and-location-header- . Even in [alternative patterns](https://stackoverflow.com/questions/5079367/use-http-status-202-for-asynchronous-operations), the accepted behavior is to return a location header with the 202 rather than the actual content. So you might expect the signature to look like `Task<CreatedAtActionResult> CreateResource`.

Reply:
OK, let's just do 200 and 201 then